Result/ResultProtocol.swift

```diff
@@ -1,30 +1,20 @@
 //  Copyright (c) 2015 Rob Rix. All rights reserved.
 
-#if swift(>=3.0)
-	public typealias ResultErrorType = ErrorProtocol
-#else
-	public typealias ResultErrorType = ErrorType
-#endif
-
 /// A type that can represent either failure with an error or success with a result value.
-public protocol ResultType {
+public protocol ResultProtocol {
```

Comment:
(Not critiquing this change though, I know it's aligned with the new stdlib conventions.)